addons/Text_Coloring/src/presenter.js

```diff
@@ -1389,5 +1380,148 @@ function AddonText_Coloring_create() {
         presenter.$wordTokens.removeClass(presenter.defaults.css.markings.wrong);
     };
 
+    presenter.setPrintableState = function(state) {
+        if (state === null || ModelValidationUtils.isStringEmpty(state))
+            return;
+
+        presenter.printableState = JSON.parse(state);
+    }
+    
+    presenter.getPrintableHTML = function (model, showAnswers) {
+        var printableHTML = '';
+        var userAnswer = presenter.getUserAnswer();
+        var didUserAnswer = userAnswer ? userAnswer.some(answer => answer.isSelected) : false;
+        var textModel = model.text.split(' ');
+        var height = model.Height / 2;
+        model = presenter.upgradeModel(model);
```

Comment:
Zamiast pracować bezpośrednio na modelu skorzystaj z validateModel i presenter.configuration (zobacz jak to drugie jest tworzone wyżej w kodzie).